plugins/woocommerce/src/Internal/EmailEditor/PageRenderer.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\EmailEditor;
+
+use MailPoet\EmailEditor\Engine\Settings_Controller;
+use MailPoet\EmailEditor\Engine\Theme_Controller;
+use MailPoet\EmailEditor\Engine\User_Theme;
+use MailPoet\EmailEditor\EmailEditorContainer;
+
+defined( 'ABSPATH' ) || exit;
+
+class PageRenderer {
+	private Settings_Controller $settingsController;
+	private Theme_Controller $themeController;
+	private User_Theme $userTheme;
+
+	public function __construct() {
+		$editorContainer = EmailEditorContainer::container();
+		$this->settingsController = $editorContainer->get( Settings_Controller::class );
+		$this->themeController = $editorContainer->get( Theme_Controller::class );
+		$this->userTheme = $editorContainer->get( User_Theme::class );
+	}
+
+	public function render() {
+		$postId = isset($_GET['post']) ? intval($_GET['post']) : 0;
+		$post = get_post($postId);
+		$currentPostType = $post->post_type;
+
+		if (!$post instanceof \WP_Post || $currentPostType !== Integration::EMAIL_POST_TYPE) {
+			return;
+		}
+
+		$emailEditorAssetsPath = WC_ABSPATH . WC_ADMIN_DIST_JS_FOLDER . 'email-editor/';
+		$emailEditorAssetsUrl = WC()->plugin_url() . '/' . WC_ADMIN_DIST_JS_FOLDER . 'email-editor/';
+
+
+		// Email editor rich text JS - Because the Personalization Tags depend on Gutenberg 19.8.0 and higher
+		// the following code replaces used Rich Text for the version containing the necessary changes.
+		$assetsParams = require $emailEditorAssetsPath . 'rich-text.asset.php';
```

Comment:
Perhaps we could improve the code by adding a check for WP and Gutenberg versions and enqueue the modified RichText script only for WP6.7 without the newer Gutenberg version.

Reply:
I would rather not remove this for the following reasons
1. This is only done for the email editor page
2. It ensures consistent with the other implementation
3. I added a [dependency](https://github.com/woocommerce/woocommerce/pull/55754/commits/c9a010081c8f3a28d53f4bbe61f996abcbfd784b#diff-189848a459676851e7c7e3afca00c5a6b45598df340444cdc2841ed31e9aef98L30) check, and this page (or package) won't be loaded for WP version less than the [recommended](https://github.com/woocommerce/woocommerce/blob/add-a-basic-integration-of-the-block-email-editor-to-the-woo-core/packages/php/email-editor/src/Engine/class-dependency-check.php#L18) version